DataFormats/Detectors/ITSMFT/MFT/src/TrackMFT.cxx

```diff
@@ -52,6 +52,77 @@ void TrackMFT::setCovariances(const SMatrix55& covariances)
   mCovariances = covariances;
 }
 
+//_________________________________________________________________________________________________
+void TrackMFT::extrapHelixToZ(double zEnd, double Field)
+{
+  /// Track extrapolated to the plane at "zEnd" considering a helix
+
+  if (getZ() == zEnd) {
+    return; // nothing to be done if same z
+  }
+
+  // Extrapolate covariances
+  extrapHelixToZCov(zEnd, Field);
+
+  // Extrapolate track parameters
+  double dZ = (zEnd - getZ()); // Propagate in meters
+  double cosphi0 = TMath::Cos(getPhi());
+  double sinphi0 = TMath::Sin(getPhi());
+  double tanl0 = getTanl();
+  double invqpt0 = getInvQPt();
+
+  double k = Field * o2::constants::math::B2C;
+  double deltax = (dZ * cosphi0 / tanl0 - dZ * dZ * k * invqpt0 * sinphi0 / (2. * tanl0 * tanl0));
+  double deltay = (dZ * sinphi0 / tanl0 + dZ * dZ * k * invqpt0 * cosphi0 / (2. * tanl0 * tanl0));
+
+  double x = getX() + deltax;
+  double y = getY() + deltay;
+  double deltaphi = +dZ * k * invqpt0 / tanl0;
```

Comment:
and avoid 4 divisons by tanl0 by precalculating it once

Reply:
done